### PR TITLE
Create tunnel client

### PR DESCRIPTION
**Pull request by alexivanov**
### Changes

This adds a client for our secure tunnels service.

This client is based on https://github.com/localtunnel/localtunnel with some modifications to make it secure and tailor it for our use case.

This package expose a command for interacting with the client but after the initial testing most of the access will be programatically via the API, e.g. within GitHub Actions or as part of other command handlers. The API will most likely change in following versions as things are still in developmens.

### Code review

Review by commit might be easier, first PR is quite large though but it's just forking the upstream master.

### CODE REVIEW

File: packages/cli/src/commands/start-local-tunnel/start-local-tunnel.command.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    localHttps: argv.localHttps,
+    localCert: argv.localCert,
+    localKey: argv.localKey,
+    localCa: argv.localCa,
+    allowInvalidCert: argv.allowInvalidCert,
+  }).catch((err) => {
+    throw err;
+  });
+
+  tunnel.on("error", (err) => {
```
Comment (dbook13) [inline packages/cli/src/commands/start-local-tunnel/start-local-tunnel.command.ts:56]: Should we call `tunnel.close()` here as well?

File: packages/cli/src/commands/start-local-tunnel/start-local-tunnel.command.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    tunnelClosedCallback.resolve(null);
+  });
+
+  logger.info(
+    `Your url is: ${tunnel.url}, user: ${tunnel.basicAuthUser}, password: ${tunnel.basicAuthPassword}`
+  );
+
+  if (argv.printRequests) {
+    tunnel.on("request", (info: IncomingRequestEvent) => {
+      console.log(new Date().toString(), info.method, info.path);
```
Comment (dbook13) [inline packages/cli/src/commands/start-local-tunnel/start-local-tunnel.command.ts:72]: logger.info instead?

File: packages/tunnel-client/src/lib/tunnel.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  tunnel_passphrase: string;
+  basic_auth_user: string;
+  basic_auth_password: string;
+}
+
+interface CreateTunnelResponseError {
+  error: string;
+}
+
+export class Tunnel extends EventEmitter {
```
Comment (dbook13) [inline packages/tunnel-client/src/lib/tunnel.ts:23]: Can you set the return types on this class' methods? With all the callbacks being passed around it's a bit confusing.

**Review by dbook13 — APPROVED** [review]
Nice!

File: packages/tunnel-client/src/lib/tunnel.ts (hunk 3 above)
Comment (alexivanov) [reply packages/tunnel-client/src/lib/tunnel.ts]: Good shout, https://github.com/alwaysmeticulous/meticulous-sdk/pull/590/commits/32c4fc16363d479e113ec6561f9241fb2da2a73e does that

**Pull request merged**